test/test_host.go

```diff
@@ -679,3 +683,140 @@ func (s *HostSuite) TestUpdateTags(t *c.C) {
 		t.Fatal("expected tag to be deleted but is still present")
 	}
 }
+
+func (s *HostSuite) TestLogSinks(t *c.C) {
+	// deploy custom logaggregator app
+	client := s.controllerClient(t)
+	logApp := &ct.App{Name: "test-logaggregator"}
+	t.Assert(client.CreateApp(logApp), c.IsNil)
+	release, err := client.GetAppRelease("logaggregator")
+	t.Assert(err, c.IsNil)
+	proc := release.Processes["app"]
+	proc.Env = map[string]string{"SERVICE_NAME": logApp.Name}
+	proc.Ports[1].Port = 55514
+	release.Processes["app"] = proc
+	release.ID = ""
+	t.Assert(client.CreateRelease(release), c.IsNil)
+	t.Assert(client.SetAppRelease(logApp.ID, release.ID), c.IsNil)
+	t.Assert(client.PutFormation(&ct.Formation{
+		AppID:     logApp.ID,
+		ReleaseID: release.ID,
+		Processes: map[string]int{"app": 1},
+	}), c.IsNil)
+	defer client.DeleteApp(logApp.ID)
+
+	// get hosts
+	hosts, err := s.clusterClient(t).Hosts()
+	t.Assert(err, c.IsNil)
+	if len(hosts) == 0 {
+		t.Fatal("no hosts found")
+	}
+
+	// wait for aggregator to come up and get address
+	instances, err := s.discoverdClient(t).Instances(logApp.Name, 10*time.Second)
+	t.Assert(len(instances), c.Equals, 1)
+
+	// add sink to controller
+	config, err := json.Marshal(&ct.SyslogSinkConfig{
+		URL:    fmt.Sprintf("syslog://%s", instances[0].Addr),
+		UseIDs: true,
+	})
+	t.Assert(err, c.IsNil)
+	sink := &ct.Sink{
+		Kind:   ct.SinkKindSyslog,
+		Config: config,
+	}
+	t.Assert(client.CreateSink(sink), c.IsNil)
+
+	var sinkAttempts = attempt.Strategy{
+		Min:   5,
+		Total: 10 * time.Second,
+		Delay: 200 * time.Millisecond,
+	}
+
+	// wait for sink to appear on host
+	err = sinkAttempts.Run(func() error {
+		sinks, err := hosts[0].GetSinks()
+		if err != nil {
+			return err
+		}
+		for _, s := range sinks {
+			if s.ID == sink.ID {
+				return nil
+			}
+		}
+		return fmt.Errorf("timed out waiting for sink to appear on host")
+	})
+	t.Assert(err, c.IsNil)
+
+	// create a test app
+	app, release := s.createApp(t)
+	defer client.DeleteApp(app.ID)
+
+	// subscribe to log messages for the test app from the test logaggregator
+	logHost, _, _ := net.SplitHostPort(instances[0].Addr)
+	logc, err := logaggc.New(fmt.Sprintf("http://%s:%d", logHost, proc.Ports[0].Port))
+	t.Assert(err, c.IsNil)
+	log, err := logc.GetLog(app.ID, &logagg.LogOpts{Follow: true})
+	t.Assert(err, c.IsNil)
+	defer log.Close()
+	msgs := make(chan *logaggc.Message)
+	stream := stream.New()
+	defer stream.Close()
+	go func() {
+		defer close(msgs)
+		dec := json.NewDecoder(log)
+		for {
+			var msg logaggc.Message
+			if err := dec.Decode(&msg); err != nil {
+				stream.Error = err
+				return
+			}
+			select {
+			case msgs <- &msg:
+			case <-stream.StopCh:
+				return
+			}
+		}
+	}()
+
+	// deploy an omni job
+	t.Assert(client.PutFormation(&ct.Formation{
+		AppID:     app.ID,
+		ReleaseID: release.ID,
+		Processes: map[string]int{"omni": 1},
+	}), c.IsNil)
+
+	// wait for syslog message from each host
```

Comment:
We should boot a 3 node cluster for this test so we know there are multiple hosts (and also prevents sending logs for other apps to the test aggregator). You'll need to use a cluster proxy to connect to the aggregator (see [here](https://github.com/flynn/flynn/blob/master/test/test_logaggregator.go#L81-L84)).